packages/graphql/src/schema/resolvers/query/fulltext.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ObjectTypeComposerFieldConfigDefinition } from "graphql-compose";
+import type { GraphQLResolveInfo } from "graphql";
+import { execute } from "../../../utils";
+import { translateRead } from "../../../translate";
+import type { Node } from "../../../classes";
+import type { Context, FulltextIndex } from "../../../types";
+import getNeo4jResolveTree from "../../../utils/get-neo4j-resolve-tree";
+import Cypher from "@neo4j/cypher-builder";
+
+export function fulltextResolver(
+    { node }: { node: Node },
+    index: FulltextIndex
+): ObjectTypeComposerFieldConfigDefinition<any, any, any> {
+    async function resolve(_root: any, args: any, _context: unknown, info: GraphQLResolveInfo) {
+        const context = createFulltextContext(index, args, _context, info);
+        const { cypher, params } = translateRead({ context, node }, node.singular);
```

Comment:
Could this full-text stuff now be seen as independent enough that it could have its own translate function? Thoughts @angrykoala?

Reply:
I did consider doing that but I barely made any changes to translate-read that were specifically for translating full-text. The changes to translate-read are almost all refactoring the sorting to use CypherBuilder. Most of the changes for full-text were needed within the functions called by translate-read. 

It's obviously possible to try and make those changes at a higher level (e.g. in a `translate-fulltext` function), but most of them depended on data that is only available within the helper functions (e.g. a field/argument name). As a result, it felt like that approach would have led to me writing lots of functions that only had minor changes compared to existing ones